src/inmanta/module.py

```diff
@@ -1657,7 +1666,7 @@ def get_relation_precedence_rules(self) -> List[RelationPrecedenceRule]:
         return [RelationPrecedenceRule.from_string(rule_as_str) for rule_as_str in self.relation_precedence_policy]
 
     def get_index_urls(self) -> List[str]:
-        return [repo.url for repo in self.repo if repo.type == ModuleRepoType.package]
+        return self.pip.index_url
```

Comment:
We must not ignore the old config: it's deprecated but still supported. In addition to that, the old and the new configs have different semantics, which is a big part of the ticket that's still missing in this PR. The old one is an index for v2 modules, the new one is an index for the entire project. The current implementation still only uses it for v2 modules, and the error message you updated above is also specific for v2 modules.

Reply:
- added support for the old config back
- log a warning if indexes are set through both options

I might be wrong but I think the indexes are used for v1 modules requirements as well [here](https://github.com/inmanta/inmanta-core/blob/be410459e1ff30671be67e829a5202dc2044b0ab/src/inmanta/module.py#L2078). As I understand it, the v1 modules are loaded above, and then requirements are collected and installed

Reply:
> I might be wrong but I think the indexes are used for v1 modules requirements as well [here](https://github.com/inmanta/inmanta-core/blob/be410459e1ff30671be67e829a5202dc2044b0ab/src/inmanta/module.py#L2078). As I understand it, the v1 modules are loaded above, and then requirements are collected and installed

You're right, it's been like that for almost a year, I had no clue.